internal/blocktx/peer_handler.go

```diff
@@ -107,18 +109,37 @@ func (ph *PeerHandler) HandleBlockAnnouncement(msg *wire.InvVect, peer p2p.PeerI
 	}
 
 	ph.blockRequestCh <- req
-
 	return nil
 }
 
 func (ph *PeerHandler) HandleBlock(wireMsg wire.Message, _ p2p.PeerI) error {
 	msg, ok := wireMsg.(*p2p.BlockMessage)
 	if !ok {
-		ph.logger.Debug(ErrUnableToCastWireMessage.Error())
 		return ErrUnableToCastWireMessage
 	}
 
 	ph.blockProcessCh <- msg
-
 	return nil
 }
+
+func extractHeightFromCoinbaseTx(tx *sdkTx.Transaction) uint64 {
+	// Coinbase tx has a special format, the height is encoded in the first 4 bytes of the scriptSig (BIP-34)
+
+	cscript := *(tx.Inputs[0].UnlockingScript)
+
+	if cscript[0] >= script.Op1 && cscript[0] <= script.Op16 { // first 16 blocks are treated differently (we have to handle it due to our tests)
+		return uint64(cscript[0] - 0x50)
+	}
```

Comment:
Would be nice to have a small single test for this case, e.g. returning height of block 9

https://whatsonchain.com/block-height/9

Reply:
1. It was added because tests only
2. We can see that it works (or not) looking at test logs
3. Actual block with height 9 is in version 1 so the code won't work for it :) 

I see no reason to add more code to code base